File: ppgs_labprog2023-main/backend/sppg/src/main/java/br/ufma/sppg/model/Tecnica.java
package br.ufma.sppg.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name="tecnica")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Tecnica {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id_tecnica")
    Integer id;

    @Column(name = "tipo")
    String tipo;

    @Column(name="titulo")
    String titulo;

    @Column(name="ano")
    Integer ano;

    @Column(name = "financiadora")
    String financiadora;

    @Column(name="outras_informacoes")
    String outrasInformacoes;

    @Column(name="qtd_grad")
    Integer qtdGrad;

    @Column(name="qtd_mestrado")
    Integer qtdMestrado;

    @Column(name="qtd_doutorado")
    Integer qtdDoutorado;
}

File: frontend/src/models/Tecnica.ts
import { Docente } from './Docente';
import { Orientacao } from './Orientacao';

export interface Tecnica {
  id: number;
  tipo: string;
  titulo: string;
  autores: string;

  ano: number;
  financiadora: string;
  outrasInformacoes: string;
  qtdGrad: string;
  qtdMestrado: string;
  qtdDoutorado: string;
  orientacoes: Orientacao[];
  docentes: Docente[];
}
